src/lib.rs

```diff
@@ -225,13 +225,34 @@ enum SystemState {
 }
 
 /// A marker trait for types which participate in the publish-subscribe system
-/// of the actor framework.
-pub trait Event: Clone + std::any::Any + 'static + Send + Sync {}
+/// of the actor framework. If it makes sense to cache your event and use it with
+/// [`Context::subscribe_and_receive_latest()`], then mark your type with [`CacheableEvent`]
+/// instead.
+pub trait Event: Clone + std::any::Any {
+    /// Internal method for the actor framework: don't implement this downstream. Make your type
+    /// implement [`CacheableEvent`] instead of [`Event`] if it makes sense to cache it.
+    #[doc(hidden)]
+    fn cache(&self) -> Option<Box<dyn std::any::Any + Send + Sync>> {
+        None
```

Comment:
In theory, letting users implement this would make it possible to condition caching on their own logic. Imagine, us saying that we want to cache only some instances of `PortalState`.

But let's keep this "private" unless we find a compelling usecase for it.

Reply:
> In theory, letting users implement this would make it possible to condition caching on their own logic.

Well, not with the current design: subsribers can only `subscribe_and_receive_latest()` to `CacheableEvent`s, and it is not possible for downstream crates to implement `Event` for anything that already implements `CacheableEvent` (it would conflict with the blanket implementation that we provide).